luigi/contrib/sge_runner.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2012-2015 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import sys
+try:
+    import cPickle as pickle
+except ImportError:
+    import pickle
+import logging
+import tarfile
+
+
+def do_work_on_compute_node(work_dir):
+
+    # Extract the necessary dependencies
+    extract_packages_archive(work_dir)
+
+    # Open up the pickle file with the work to be done
+    os.chdir(work_dir)
+    with open("job-instance.pickle", "r") as f:
+        job = pickle.load(f)
+
+    # Do the work contained
+    job.work()
+
+
+def extract_packages_archive(work_dir):
+    package_file = os.path.join(work_dir, "packages.tar")
+    if not os.path.exists(package_file):
+        return
+
+    curdir = os.path.abspath(os.curdir)
+
+    os.chdir(work_dir)
+    tar = tarfile.open(package_file)
+    for tarinfo in tar:
+        tar.extract(tarinfo)
+    tar.close()
+    if '' not in sys.path:
+        sys.path.insert(0, '')
+
+    os.chdir(curdir)
+
+
+def main(args=sys.argv):
```

Comment:
Wait, why do we have a file with a `main()` in it? Don't all sge related logic fit in the Task?